magefile.go

```diff
@@ -0,0 +1,288 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//+build mage
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+
+	"github.com/elastic/go-txfile/dev-tools/lib/mage/gotool"
+	"github.com/elastic/go-txfile/dev-tools/lib/mage/mgenv"
+	"github.com/elastic/go-txfile/dev-tools/lib/mage/xbuild"
+)
+
+// Info namespace is used to print additional docs, help messages, and other info.
+type Info mg.Namespace
+
+// Prepare namespace is used to prepare/download/build common depenendencies for other tasks to run.
+type Prepare mg.Namespace
+
+// Check runs pre-build checks on the environment and source code. (e.g. linters)
+type Check mg.Namespace
+
+// Build namespace defines the set of build targets
+type Build mg.Namespace
+
+const buildHome = "build"
+
+// environment variables
+var (
+	envBuildOS    = mgenv.String("BUILD_OS", runtime.GOOS, "(string) set compiler target GOOS")
+	envBuildArch  = mgenv.String("BUILD_ARCH", runtime.GOARCH, "(string) set compiler target GOARCH")
+	envTestUseBin = mgenv.Bool("TEST_USE_BIN", false, "(bool) reuse prebuild test binary when running tests")
+	envTestShort  = mgenv.Bool("TEST_SHORT", false, "(bool) run tests with -short flag")
+	envFailFast   = mgenv.Bool("FAIL_FAST", false, "(bool) do not run other tasks on failure")
+)
+
+var xProviders = xbuild.NewRegistry(map[xbuild.OSArch]xbuild.Provider{
+	xbuild.OSArch{"linux", "arm"}: &xbuild.DockerImage{
+		Image:   "balenalib/revpi-core-3-alpine-golang:latest-edge-build",
+		Workdir: "/go/src/github.com/elastic/go-txfile",
+		Volumes: map[string]string{
+			filepath.Join(os.Getenv("GOPATH"), "src"): "/go/src",
+		},
+	},
+})
+
+// targets
+
+// Env prints environment info
+func (Info) Env() {
+	printTitle("Mage environment variables")
+	for _, k := range mgenv.Keys() {
+		v, _ := mgenv.Find(k)
+		fmt.Printf("%v=%v\n", k, v.Get())
+	}
+	fmt.Println()
+
+	printTitle("Go info")
+	sh.RunV(mg.GoCmd(), "env")
+	fmt.Println()
+
+	printTitle("docker info")
+	sh.RunV("docker", "version")
+}
+
+// Vars prints the list of registered environment variables
+func (Info) Vars() {
+	for _, k := range mgenv.Keys() {
+		v, _ := mgenv.Find(k)
+		fmt.Printf("%v=%v  : %v\n", k, v.Default(), v.Doc())
+	}
+}
+
+// All runs all Prepare tasks
+func (Prepare) All() { mg.Deps(Prepare.Dirs) }
+
+// Dirs creates requires build directories for storing artifacts
+func (Prepare) Dirs() error { return mkdir("build") }
+
+// Lint runs golint
+func (Check) Lint() error {
+	return errors.New("TODO: implement me")
```

Comment:
Still noop. Checks will be added in the future, so to not break the build right now.